src/TypeScript/NodeJSLocator/NodeVersionInfoProvider.cs

```diff
@@ -71,23 +65,8 @@ public string Locate()
 
                 var nodeVersion = getNodeExeVersion(nodeExePath);
 
-                if (!IsCompatibleVersion(nodeVersion))
-                {
-                    logger.WriteLine(Resources.ERR_IncompatibleVersion, nodeVersion, nodeExePath);
-                    continue;
-                }
-
-                logger.WriteLine(Resources.INFO_FoundCompatibleVersion, nodeVersion, nodeExePath);
-                return nodeExePath;
+                yield return new NodeVersionInfo(nodeExePath, nodeVersion);
```

Comment:
yield return so it doesn't degrade the performance of eslint-bridge initialization. We will still stop as soon as we find the first compatible.

Reply:
It's a non-obvious perf optimisation, so it's worth commenting in the code.